### ISSUE TITLE
avm2: Gamezhero preloader does not work

### ISSUE THREAD

**Issue by n0samu** [ruffle-rs/ruffle#9507]
### Describe the bug

AVM2 games that use the Gamezhero preloader show only a blank screen and do not load. Affected games include [Where is My Beard](https://www.kongregate.com/games/gamezhero/where-is-my-beard) (#1956), [Back to Alien Party](https://www.gamezhero.com/games/backtoalienparty), and [Catchy Orbit](https://www.newgrounds.com/portal/view/630301/format/flash?emulate=flash).

The error is:
`ERROR run_frame:run_all_phases_avm2: ruffle_core::display_object::movie_clip: Got RustError("ArgumentError: Cannot remove object from display list it is not a child of.") when constructing AVM2 side of movie clip of type Preloader`

### Expected behavior

The preloader screen should display and the Play button should appear.

### Affected platform

Desktop app

### Operating system

Windows 10

**Comment by Lord-McSweeney** [comment]
Still happening on latest nightly.

**Comment by n0samu** [comment]
With #10697, the games no longer show the above error, but they don't appear to load any farther. Where's My Beard and Catchy Orbit show this new error:
```
ERROR run_frame:run_all_phases_avm2: ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "api_ini_engine", class: com.gameallianz.api.as3::GameAllianzApiEvent, ptr: 0x1d41e036490 }) to handler FunctionObject(FunctionObject { ptr: 0x1d4122a3c80 }) : ReferenceError: Error #1069: Property flash.printing::PrintJob not found on global and there is no default value.
        at com.gameallianz.engine.as3::Engine/action()
        at Function/apply()
        at com.gameallianz.api.as3.classes::GameAllianzApiExtended$/_-2D()
        at Function/apply()
        at com.gameallianz.api.as3.classes::GameAllianzApiExtended$/action()
        at Function/apply()
        at com.gameallianz.api.as3.classes::GameAllianzApiExtended$/_-9M()
        at com.gameallianz.api.as3.classes::GameAllianzApiExtended$/_-f()
        at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
        at com.gameallianz.api.as3.classes::GameAllianzApiExtended$/dispatchEvent()
        at com.gameallianz.api.as3.classes::GameAllianzApiExtended$/_-AD()
        at com.gameallianz.api.as3.classes::GameAllianzApiExtended$/_-2U()
```

**Comment by n0samu** [comment]
Also seeing these errors:
`ERROR ruffle_core::tag_utils: Error running definition tag: DefineVideoStream, got Couldn't read SWF`
`WARN run_frame: ruffle_core::display_object::movie_clip: DoAction tag in AVM2 movie`

Hopefully this will be progressed by #10756 and #11005.

**Comment by n0samu** [comment]
Seems like this was fixed by #10756 🎉
